pkg/ingester/ingester.go

```diff
@@ -1833,13 +1846,42 @@ func (i *Ingester) openExistingTSDB(ctx context.Context) error {
 	queue := make(chan string)
 	group, groupCtx := errgroup.WithContext(ctx)
 
+	userIDs, err := i.getAllTSDBUserIDs()
+	if err != nil {
+		level.Error(i.logger).Log("msg", "error while finding existing TSDBs", "err", err)
+		return err
+	}
+
+	if len(userIDs) == 0 {
+		return nil
+	}
+
+	var concurrentOpenTSDBCount = i.cfg.BlocksStorageConfig.TSDB.DeprecatedMaxTSDBOpeningConcurrencyOnStartup
```

Comment:
[design] I would move all this logic in a dedicated standalone function (taking in input `i.cfg.BLocksStorageConfig`) which then you can unit test.